### DISCUSSION TITLE
MonitorControl v3.0.0

### DISCUSSION THREAD

**Discussion by JoniVR** [MonitorControl/MonitorControl#569] [Release Discussion]
Full release notes for 3.0.0

### Enhancements
- Added **support for Apple Silicon**
- Updated application icon.
- Replaced menu bar slider titles with icons (only on macOS Big Sur and above).
- New Big Sur style menu on Big Sur and above
- Added an "About" tab.
- New "Displays" tab in preferences (this combines the functions of the old "Display" and "Advanced" tab).
- Enable software dimming if required, this is enabled by default.
- Hide OSD function hides the mac volume OSD instead of trying to suppress LG display overlay.
- Show an OSD lock symbol when keyboard change is not allowed for a few seconds after sleep or display reconfiguration.
- Keyboard controls now works when menu is open.
- Active screen is now determined solely by the location of the mouse cursor (no need to click or anything). This also solves the issue regarding full screen apps which in the past resulted in the wrong screen being detected as active.
- Added "Enable Mute DDC command" to handle displays that do or do not support this command (Thanks to @robertbressi)
- In Settings/Display the 'DDC' checkbox can now be unchecked - this combined with 'Enable software control...' will let misbehaving displays be controlled via software dimming. This might also be a better alternative for OLED or microLED displays.
- Full compatibility with custom expert ColorSync Display Calibration Profiles.
- Subtle OSD effect when using 'Further lower brightness via software shading'
- Upon configuration change, when previously saved software dimming values are restored, a subtle OSD effect will give a hint to the user that MonitorControl is doing this.
- Upon configuration change, restoring software dimming is smoothed out to be less disruptive.
- Moved "Change Brightness/Volume for all screens" under Keyboard in Settings.
- Added software darkening option in place of ‘lower contrast after brightness’.
- Added software brightness control as a fallback if DDC is not available.
- Added Hide Menu Icon option.
- Displays that has mirrored screens are controlled along with mirror master.
- OSD is shown on the mirror master screen when a controllable display is shadowed by an uncontrollable display.
- Display naming reflects mirroring better.
- Better handling of rapid configuration changes (when user closes/opens lid or plugs in and out displays rapidly).
- Submenu mode looks better now and is activated only with 3 or more displays.
- Proper support for (future) Apple Silicon Macs with multiple external display support.
- Better handling of runtime changes in display configuration.
- Added option to disable volume slider.
- DisplayServices framework bridging instead of existing Swift workaround.
- Major refactoring, moved most arm64 related stuff to a single self-contained utility class.
- Lazy display update (upon configuration changes the OS sent 4-8 notifications to the app which made updating the display configuration costly - the app now waits 2 seconds until things settle).
- Better handling of mirrors (displays that are mirror targets are now detected by the app).
- Fixed Display and system sleep issues.
- MonitorControl now disengages from media keys when there is no external display connected.
- Better handling of virtual screens (AirPlay, Sidecar).
- Display naming changes (now consistent with System Preferences).
- Detection and blocking of M1 Mac Mini (Macmini9,1) HDMI port to prevent issues.
- Updated/Added Russian, Chinese-Simplified, Chinese-Taiwan, Dutch, German, Polish, Spanish, Turkish, Polish, Hungarian and Korean localisations 
(Thanks to everyone helping to translate!)

### Bug Fixes
- Fixed Intel DDC Swift optimisation issue (#562)
- Fixed not working after sleep mode for some on Apple Silicon (#530)
- Fixed some LG and Samsung displays having problems with Mute (improved 'Enable Mute DDC command') - (#170)
- Fixed app not working with multiple identical monitors on Intel (#49)
- Fix to make sure DDC communications don't happen in parallel when both slider menu and keyboard is used (this might have caused problems with some docks with multiple display outputs).
- Fixed volume control feedback audio (clicking sound) during key repeat (it should play on keyup only as this is the macOS standard).
- Fixed duplication of volume control feedback audio if there are multiple external displays and 'Change... for all screens' is enabled.

### Notes
- **We're still looking for some updated translations, if you can help out, please do!** 
- Thanks to everyone who helped testing, contributed and donated! It really helps a lot! 
- Special thanks to @waydabber for making this absolutely amazing release happen! 🔥 🔥 🚀 

<hr /><em>This discussion was created from the release <a href='https://github.com/MonitorControl/MonitorControl/releases/tag/v3.0.0'>MonitorControl v3.0.0</a>.</em>

**Comment by tedd4u** [comment]
@JoniVR & @waydabber -- y'all rock 👍. Thank you so much for bringing this to M1 and all the other enhancements. What an epic update cycle. I hope others will kick in a few $$ like I did: https://opencollective.com/monitorcontrol

**Reply by waydabber** [reply]
Thanks @tedd4u for the kind works and the contribution to the project! :)

**Reply by MattCowgill** [reply]
I've donated too. Thank you @JoniVR  and @waydabber for all your work

**Reply by Sayvai** [reply]
Donated, and well deserved. A small yet powerful utility that does wonders for our health, and especially for our eyes 👀

**Comment by sysatimi** [comment]
Thanks so much for doing this. I came across the older version several months ago, but it did not work with M1 Macs so I was saddened. Now it does, and it is brilliant!

One observation I have with my particular configuration is that as I use the F1/F2 keys to adjust brightness on a Dell U4021Q display, brightness increases for the first 12 steps, then 13-16 all set the screen to dim to the level of about step 8. So I get something like:

           12
          11
         10
        9
       8        13  14  15  16
      7
     6
    5
   4
  3
 2
1

Wondering if anyone has noticed something like this?

MacBookPro 13 inch M1
Dell U4021Q Display

**Reply by waydabber** [reply]
Hi @sysatimi - this happens if the display incorrectly reports the control range (max. value) for the brightness. It might report that the max. brightness is 150 while it is actually 100 and between 100-150 it defaults to 50%. You can try to set polling to `None` in the Advanced section under Displays. This will make MonitorControl treat the maximum as 100 (which is correct for most displays). If this does not help then there is a planned feature which will let you manually set the maximum (and minimum) brightness value for the display - https://github.com/MonitorControl/MonitorControl/issues/310

**Reply by waydabber** [reply]
Hi @sysatimi 

In 3.1.0 you'll be able to manually adjust the DDC range for problematic displays (you can have this tody, but you have to build the app manually from the `3.1.0` branch, see build instructions).

<img width="803" alt="133389131-49835e45-4527-4bdb-9de4-b47a74362aab" src="https://user-images.githubusercontent.com/37590873/133392472-620029cd-5de2-45f2-829d-681d2670e0ee.png">

**Comment by sysatimi** [comment]
Tried that, but it made no difference to the Dell display behaviour. It did dim my MacBook screen and change the alignment between brightness ‘pips’ between the two displays. However returning polling to normal set things right again.

Regards,


Paul Fieber
IT Director
[Atimi Software Inc.]<http://www.atimi.com/>
Premium App Development
www.atimi.com<http://www.atimi.com/>  |  @atimisoftware<https://twitter.com/AtimiSoftware>
direct.
778-372-2805
mobile.
778-229-0023
office.
778-372-2800
email.
***@***.******@***.***>

Suite 800 - 800 W Pender St, Vancouver, BC   V6C 2V6

Atimi is CASL compliant. If you no longer wish to be contacted by us through electronic messages, you may opt-out here.<http://www.atimi.com/contact/unsubscribe/>



From: waydabber ***@***.***>
Date: Tuesday, September 7, 2021 at 1:18 PM
To: MonitorControl/MonitorControl ***@***.***>
Cc: System Administrator ***@***.***>, Mention ***@***.***>
Subject: Re: [MonitorControl/MonitorControl] MonitorControl v3.0.0 (#569)

Hi @sysatimi<https://github.com/sysatimi> - this happens if the display incorrectly reports the control range (max. value) for the brightness. It might report that the max. brightness is 150 while it is actually 100 and between 100-150 it defaults to 50%. You can try to set polling to None in the Advanced section under Displays. This will make MonitorControl treat the maximum as 100 (which is correct for most displays). If this does not help then there is a planned feature which will let you manually set the maximum (and minimum) brightness value for the display - #310<https://github.com/MonitorControl/MonitorControl/issues/310>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/MonitorControl/MonitorControl/discussions/569#discussioncomment-1293087>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQV6Q2SVZHAJJPUSAE352O3UAZXS7ANCNFSM5DSVDFCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by waydabber** [reply]
Great, glad that it works (if I understand correctly). The fact that with polling set to 'None' changes the range (I think this is what you mean by "change the alignment between brightness ‘pips’ between the two displays") means that the display do not use the mostly standard 0-100 scale internally and based on the previous issue it might not report these values properly (or there is something wrong with MC if the scale is non-standard). In an upcoming release you'll be able to fine tune it by specifying custom bounds (#310) fixing this, but this is not possible in the current release.

**Comment by micbblom** [comment]
Thank you very much! Greatly appreciate the work. Please allow me an off-topic question. On M1 Mac Mini I cannot use DDC with HDMI. Will I be able to use a USB-C to Displayport or a USB-C to HDMI cable to use DDC with my monitor? Thanks, Michael

**Reply by waydabber** [reply]
Hi @micbblom, yes, that should work - USB-C to DisplayPort dongles/cables usually work without exception, the majority of USB-C to HDMI cables  also work but might depend on the IC inside and if it uses DisplayPort Alt Mode internally or HDMI Alt Mode.

**Comment by alexandrosraikos** [comment]
Just wanted to add another voice to the choir saying "Thanks"!
This little utility has made life so much easier and this version has so many nice additions and changes. 🎊

**Comment by kaeferfreund** [comment]
MANY, MANY THANKS!!! This is really a live saver, as coming from intel macs the DDC did not work the way they used to. 

Great work!

**Comment by volibal23** [comment]
HDMI->HDMI from M1 Mac is not possible to control the volume ?
I don't have the option to control the volume :( only on the USB-C Display.

**Reply by waydabber** [reply]
Hi, sorry to say but the HDMI port on the Mac Mini (both M1 and the 2018 Intel) blocks DDC communication. This seems to be a hardware or low level driver limitation which Apple put in place for unknown reasons (or simply nobody was able to figure out how to do it so far). You'll have to use the USB-C port to control your display (most USB-C HDMI dongles work fine). This is an especially painful limitation on the M1 mini as it limits options for dual display configs.

**Reply by volibal23** [reply]
This is fucking shit :( thanks for the information :)

**Reply by volibal23** [reply]
M1 Mac HDMI-> DispalyPort Cable (Display) will have the same effect, or ?

**Reply by micbblom** [reply]
I fixed it with USB-C -> Displayport. Afterwards I donated.

**Reply by waydabber** [reply]
> M1 Mac HDMI-> DispalyPort Cable (Display) will have the same effect, or ?

Yes, most likely it will not work either. The app actually blocks the M1 mini HDMI port as trying to communicate through it via DDC causes problems (this block can be easily removed if you want to experiment with it and build a custom version, I'll tell you where).

**Reply by volibal23** [reply]
I want to experiment it :)

**Reply by waydabber** [reply]
Hi @volibal23,

you can try to force enabling it but you need to change the code and build the app for you:

You can simply remove lines 275-277 here and build the app in XCode by following the build instructions.

https://github.com/MonitorControl/MonitorControl/blob/6bba2f6ed749afd0b7da32c0769d01bf9a4ef25e/MonitorControl/Support/Arm64DDC.swift#L275

Please note however that others tried to do this and did not succeed.

If you want to experiment with it further, you might want to try to send custom DDC commands and change the I2C communication addresses as described here: https://github.com/waydabber/m1ddc/issues/3#issuecomment-908513231

You can modify m1ddc's source code for these. In the following line you can try changing 0x37 and 0x51 and see if something maybe works (?) on the HDMI port:

https://github.com/waydabber/m1ddc/blob/a23b9ce15675c2b18751b27b6d9db26bd8b2b87f/m1ddc.m#L286
